autorest/validations/validations.go

```diff
@@ -0,0 +1,415 @@
+/*
+Package validations provides methods for validating parameter value using reflection.
+*/
+package validations
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint stores constraint name, target field name
+// Rule and chain validations.
+type Constraint struct {
+
+	// Target field name for validation.
+	Target string
+
+	// Constraint name e.g. Minlength, Maxlength, Pattern, etc.
+	Name int
```

Comment:
This change to int because I am using iota for constants now. It was of string type with string constants earlier.